### PR TITLE
fix bug for issues 879

### PR DESCRIPTION
**Pull request by kai8406**
1.Correctly obtain redis db object
2.Correctly calculate ResponseTime

### CODE REVIEW

File: agent/agent-plugins/redis-jedis-2.x/src/main/java/org/bithon/agent/plugin/redis/jedis2/interceptor/OnCommand.java [hunk 1]
```diff
@@ -50,7 +49,7 @@ public class OnCommand extends AroundInterceptor {
     public InterceptionDecision before(AopContext aopContext) throws Exception {
         Jedis jedis = aopContext.getTargetAs();
         String hostAndPort = jedis.getClient().getHost() + ":" + jedis.getClient().getPort();
-        long db = jedis.getDB();
+        long db = jedis.getClient().getDB();
```
Comment (FrankChen021) [inline agent/agent-plugins/redis-jedis-2.x/src/main/java/org/bithon/agent/plugin/redis/jedis2/interceptor/OnCommand.java:52]: how did the original code cause the NoClassDefFoundError?

File: agent/agent-plugins/redis-jedis-2.x/src/main/java/org/bithon/agent/plugin/redis/jedis2/interceptor/OnCommand.java [hunk 2]
```diff
@@ -59,17 +58,15 @@ public InterceptionDecision before(AopContext aopContext) throws Exception {
 
         if (!ignoreCommands.contains(command)) {
             ITraceSpan span = TraceContextFactory.newSpan("jedis");
-            if (span == null) {
-                return InterceptionDecision.SKIP_LEAVE;
+            if (span != null) {
```
Comment (FrankChen021) [inline agent/agent-plugins/redis-jedis-2.x/src/main/java/org/bithon/agent/plugin/redis/jedis2/interceptor/OnCommand.java:61]: this change is a good catch. Returning SKIP_LEAVE in previous release indeed cause inacurate metrics 👍

File: agent/agent-plugins/redis-jedis-3.x/src/main/java/org/bithon/agent/plugin/redis/jedis3/interceptor/OnCommand.java [hunk 3]
```diff
@@ -82,9 +78,10 @@ public void after(AopContext aopContext) {
             span.tag(aopContext.getException()).finish();
         }
 
-        JedisContext ctx = InterceptorContext.remove("redis-command");
+        JedisContext ctx = InterceptorContext.getAs("redis-command");
```
Comment (FrankChen021) [inline agent/agent-plugins/redis-jedis-3.x/src/main/java/org/bithon/agent/plugin/redis/jedis3/interceptor/OnCommand.java:81]: why is this change needed?

Comment (kai8406) [reply agent/agent-plugins/redis-jedis-3.x/src/main/java/org/bithon/agent/plugin/redis/jedis3/interceptor/OnCommand.java:81]: If remove is used, the InterceptorContext will no longer have the context information related to redis-command. Therefore, it needs to be modified to obtain the redis-command context based on the key, and then remove it after the statistics are completed.

File: agent/agent-plugins/redis-jedis-2.x/src/main/java/org/bithon/agent/plugin/redis/jedis2/interceptor/OnCommand.java (hunk 1 above)
Comment (FrankChen021) [reply agent/agent-plugins/redis-jedis-2.x/src/main/java/org/bithon/agent/plugin/redis/jedis2/interceptor/OnCommand.java:52]: The change looks good to me because Line 51 also uses the `getClient` method to access some information.

what I don't understand is that inside the `jedis.getDB()`, it calls `getClient().getDB()`, which looks the same as `jedis.getClient().getDB()`.

Could you upload your test project to GitHub so that I can investigate more if there's sth we can improve inside the class loader?

File: agent/agent-plugins/redis-jedis-3.x/src/main/java/org/bithon/agent/plugin/redis/jedis3/interceptor/OnCommand.java (hunk 3 above)
Comment (FrankChen021) [reply agent/agent-plugins/redis-jedis-3.x/src/main/java/org/bithon/agent/plugin/redis/jedis3/interceptor/OnCommand.java:81]: The `remove` removes the object from current thread local and returns the context object, I think it's the same with  the code you are changing. do I miss sth?
